File: reference/dotnet/Domain/Company.Product.Domain.UseCases/Exceptions/NotAuthorizedException.cs
using System;

namespace Company.Product.Domain.UseCases.Exceptions
{
    public class NotAuthorizedException : Exception
    {
        public NotAuthorizedException()
           : this(null) { }

        public NotAuthorizedException(Exception innerException)
            : base("User is not authorized to execute this use-case.", innerException) { }
    }
}

File: reference/springboot/domain-usecases/src/main/java/org/example/domain/usecases/exceptions/NotAuthorizedException.java
package org.example.domain.usecases.exceptions;

public class NotAuthorizedException extends RuntimeException {

    protected NotAuthorizedException() {
        this(null);
    }

    protected NotAuthorizedException(Throwable cause) {
        super("User is not authorized to execute this use-case.", cause);
    }
    
}
